### PR TITLE
VACMS-15001 Add PACT Act test plan

### PR DESCRIPTION
**Pull request by randimays**
Ticket: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15001

We are adding a draft test plan for the PACT Act to facilitate staging review when the time comes.

### CODE REVIEW

File: products/pact-act-wizard/engineering/README.md [hunk 1]
```diff
@@ -1 +1,83 @@
...
+Both utilities are thoroughly unit tested [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/pact-act/tests/utils.unit.spec.js).
+
+#### Unit Test Coverage Statistics
+|Application (src/applications) | Lines | Functions | Statements | Branches |
+|---|---|---|---|---|
+|pact-act | TBD | TBD | TBD | TBD |
+
+### Cypress Test Coverage
+
+There is a comprehensive [Cypress test suite](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/income-limits/tests/pact-act.cypress.spec.js). It tests the following scenarios:
```
Comment (chriskim2311) [inline products/pact-act-wizard/engineering/README.md:73]: Should this be` /pact-act/tests/pact-act.cypress.spec.js`

**Review by chriskim2311 — APPROVED** [review]
LGTM I agree majority of testing will be with cypress and one small comment

File: products/pact-act-wizard/engineering/README.md [hunk 2]
```diff
@@ -1 +1,83 @@
...
+`*` 1. Selection of one or more geographic locations OR<br/>
+`*` 2. "None of these locations" or "I'm not sure" response
+
+## Automated Test Coverage
+### Unit Test Coverage
+#### Shadow DOM Limitations
+
+Unit testing PACT Act code had its challenges, as the shadow DOM cannot be targeted in a unit test. 
+
+Any actions requiring `<va-button>` clicks could not be tested as the actual `<button>` is contained in the shadow DOM. Our Back and Continue buttons are `<va-button>`s, so we could not simulate clicking them for form validation or navigation forward or backward.
```
[the remark this answers is not in the document]
Comment (randimays) [reply products/pact-act-wizard/engineering/README.md:55]: Good call out. I updated the language in this section and the manual testing section above. The DST knows about the shadow DOM implications with testing. There are some less-than-ideal workarounds, but personally I think Cypress is just fine for covering these things, provided our Cypress suite doesn't get too heavy.

**Review by jilladams — APPROVED** [review]

**Review by wesrowe — COMMENTED** [review]
The content looks pretty good to me for this stage.

The past tense bothers me a little – I think present tense would work in most places and would make it feel a little less attached to a moment in time (which hasn't arrived yet). 

Also, it looks like you've added this as the ReadMe. If that's the case... it doesn't feel like a ReadMe to me. But if this doesn't hurt the findability then I'm okay either way.

**Comment by randimays** [comment]
@wesrowe Updated to future tense and moved this to a `testing-qa.md` file 👍

File: products/pact-act-wizard/engineering/testing-qa.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  - "No" or "I'm not sure" selected for each question
+  - Combinations of the above
+  - Results screen accordions are validated for proper open/close behavior as well as content
+- Validation errors as expected
+  - Yes/No/I'm not sure questions - one response must be selected to continue
+  - Multi-checkbox questions - at least one response must be selected to continue`*`
+- Deep linking prevention
+  - If a user lands within the flow for any reason, they are redirected to the /introduction screen.
+
+`*` For multi-checkbox questions that have a "None of these locations" or "I'm not sure" response, we will accept only:<br/>
```
Comment (wesrowe) [inline products/pact-act-wizard/engineering/testing-qa.md:45]: Randi, there is late-breaking news on these questions – we will no longer have weird checkbox logic like this. All questions with a No or Not Sure response option will be radio buttons! so you can delete this bit

**Review by wesrowe — COMMENTED** [review]
I left one comment. Otherwise looks good!

File: products/pact-act-wizard/engineering/testing-qa.md (hunk 3 above)
Comment (randimays) [reply products/pact-act-wizard/engineering/testing-qa.md:45]: Thanks @wesrowe!

**Pull request merged**